ipld/resolver/src/observe.rs

```diff
@@ -0,0 +1,331 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: MIT
+use ipc_observability::{
+    impl_traceable, impl_traceables, register_metrics, Recordable, TraceLevel, Traceable,
+};
+use lazy_static::lazy_static;
+use libipld::cid::Cid;
+use libp2p::gossipsub::TopicHash;
+use libp2p::PeerId;
+use prometheus::{
+    register_histogram, register_int_counter, register_int_gauge, Histogram, IntCounter, IntGauge,
+    Registry,
+};
+use std::time::Duration;
+
+register_metrics! {
+    IPLD_RESOLVER_PING_RTT: Histogram =
+        register_histogram!("ipld_resolver_ping_rtt", "Ping roundtrip time");
+
+    IPLD_RESOLVER_PING_TIMEOUT: IntCounter =
+        register_int_counter!("ipld_resolver_ping_timeouts", "Number of timed out pings");
```

Comment:
I think we could merge these together and have a label instead? Instead of having 3 metrics it can be one with status label Same goes for others (only if it makes sense in the given context). What do you think?

Reply:
Overloading as a label will require secondary indexing and will be counter productive if per-event counters are wanted. I think they are wanted, but I don't have a strong opinion about it.

Reply:
I just didn’t want to generate a lot of similar metrics. I don’t think we gained enough experience debugging the resolver based on these, so it’s a tough call. We can leave it as it is for now.